src/hotspot/share/prims/jvm.cpp

```diff
@@ -3282,6 +3282,8 @@ JVM_END
 
 JVM_ENTRY(jboolean, JVM_ReferenceRefersTo(JNIEnv* env, jobject ref, jobject o))
   oop ref_oop = JNIHandles::resolve_non_null(ref);
+  // PhantomReference has it's own implementation of refersTo().
+  assert(!java_lang_ref_Reference::is_phantom(ref_oop), "precondition");
```

Comment:
Maybe this would be even clearer if it said:
```
// PhantomReference has it's own implementation of refersTo(). See: JVM_PhantomReferenceRefersTo
```